AuthExampleIT.testAuthApiKey test failing

`AuthExampleIT.testAuthApiKey` was [failing](https://btx.cloud.google.com/invocations/94221529-f752-4b64-bdb7-3f3ed45f4d7c/targets/cloud-devrel%2Fjava%2Fjava-docs-samples%2Fjava11%2Fpresubmit;config=default/log) with an error " INVALID_ARGUMENT: API key expired". As requested by @ldetmer, the test has been temporarily disabled in PR #9994  to unblock the PR. Please address the issue and re-enable the test.

```
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.841 s <<< FAILURE! - in com.google.cloud.auth.samples.AuthExampleIT
[ERROR] com.google.cloud.auth.samples.AuthExampleIT.testAuthApiKey  Time elapsed: 2.679 s  <<< ERROR!
com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: API key expired. Please renew the API key.
Caused by: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: API key expired. Please renew the API key.

[ERROR] Errors: 
[ERROR]   AuthExampleIT.testAuthApiKey » InvalidArgument io.grpc.StatusRuntimeException:...
[ERROR] Tests run: 12, Failures: 0, Errors: 1, Skipped: 0
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-M5:verify (default) on project auth: There are test failures.
```

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -61,12 +61,10 @@ public class AuthExampleIT {
     assertTrue(output.contains("Buckets:"));
   }
 
-  @Ignore("Temporarily disabled due to failing test (Issue #10023).")
   @Test
   public void testAuthApiKey() throws IOException, IllegalStateException {
-    //TODO: Re-enable this test after fixing issue #10023.
     String projectId = ServiceOptions.getDefaultProjectId();
-    String keyDisplayName = "Test API Key";
+    String keyDisplayName = "Test API Key " + System.currentTimeMillis();;
     String service = "language.googleapis.com";
     String method = "google.cloud.language.v2.LanguageService.AnalyzeSentiment";
     Key apiKey = null;

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -20,9 +20,9 @@ import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 import com.google.api.apikeys.v2.Key;
+import com.google.api.gax.rpc.InvalidArgumentException;
 import com.google.cloud.ServiceOptions;
 import io.grpc.StatusRuntimeException;
-import com.google.api.gax.rpc.InvalidArgumentException;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.PrintStream;
@@ -66,15 +66,12 @@ public class AuthExampleIT {
   @Test
   public void testAuthApiKey() throws IOException, IllegalStateException {
     String projectId = ServiceOptions.getDefaultProjectId();
-    java.util.Random random = new java.util.Random();
-    String keyDisplayName = "Test API Key " + random.nextInt();
+    String keyDisplayName = "Test API Key";
     String service = "language.googleapis.com";
     String method = "google.cloud.language.v2.LanguageService.AnalyzeSentiment";
     Key apiKey = null;
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
-      assertNotNull(apiKey);
-      System.out.println("key string " + apiKey.getKeyString());
       String output = authenticateUsingApiKeyWithRetry(apiKey.getKeyString());
       assertTrue(output.contains("magnitude:"));
     } finally {
@@ -92,7 +89,7 @@ public class AuthExampleIT {
     for (int i = 0; i < retries; i++) {
       try {
         return ApiKeyAuthExample.authenticateUsingApiKey(apiKey);
-      } catch (StatusRuntimeException | InvalidArgumentException e ) {
+      } catch (StatusRuntimeException | InvalidArgumentException e) {
         if (e.getMessage().contains("API key expired")) {
           System.out.println("API key not yet active, retrying...");
           try {

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -95,7 +95,7 @@ public class AuthExampleIT {
           System.out.println("API key not yet active, retrying...");
           try {
             Thread.sleep(delay);
-          } catch (InterruptedException ignored) {}
+          } catch (InterruptedException ignored) { }
         } else {
           throw e;
         }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -72,12 +72,13 @@ public class AuthExampleIT {
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
       assertNotNull(apiKey);
+      assert
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
 
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
-        AuthTestUtils.deleteTestApiKey(apiKey.getName());
+       // AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }
   }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -72,11 +72,12 @@ public class AuthExampleIT {
     Key apiKey = null;
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
+
       String output = authenticateUsingApiKeyWithRetry(apiKey.getKeyString());
+
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
-        System.out.println("trying to delete " + apiKey.getKeyString());
         AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -22,6 +22,7 @@ import static org.junit.Assert.assertTrue;
 import com.google.api.apikeys.v2.Key;
 import com.google.cloud.ServiceOptions;
 import io.grpc.StatusRuntimeException;
+import com.google.api.gax.rpc.InvalidArgumentException;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.PrintStream;
@@ -91,12 +92,14 @@ public class AuthExampleIT {
     for (int i = 0; i < retries; i++) {
       try {
         return ApiKeyAuthExample.authenticateUsingApiKey(apiKey);
-      } catch (StatusRuntimeException e) {
+      } catch (StatusRuntimeException | InvalidArgumentException e ) {
         if (e.getMessage().contains("API key expired")) {
           System.out.println("API key not yet active, retrying...");
           try {
             Thread.sleep(delay);
-          } catch (InterruptedException ignored) { }
+          } catch (InterruptedException ignored) {
+            // ignore iterrupted exception and retry test
+          }
         } else {
           throw e;
         }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -71,13 +71,11 @@ public class AuthExampleIT {
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
       assertNotNull(apiKey);
-      System.out.println("created key " + apiKey.getKeyString());
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
 
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
-        System.out.println("deleting key " + apiKey.getKeyString());
         AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -70,12 +70,13 @@ public class AuthExampleIT {
     Key apiKey = null;
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
-
+      System.out.println("created key " + apiKey.getKeyString());
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
 
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
+        System.out.println("deleting key " + apiKey.getKeyString());
         AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -70,7 +70,7 @@ public class AuthExampleIT {
     Key apiKey = null;
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
-      assertNotNull(apiKey)
+      assertNotNull(apiKey);
       System.out.println("created key " + apiKey.getKeyString());
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
 

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -64,7 +64,8 @@ public class AuthExampleIT {
   @Test
   public void testAuthApiKey() throws IOException, IllegalStateException {
     String projectId = ServiceOptions.getDefaultProjectId();
-    String keyDisplayName = "Test API Key " + System.currentTimeMillis();;
+    java.util.Random random = new java.util.Random();
+    String keyDisplayName = "Test API Key " + random.nextInt();
     String service = "language.googleapis.com";
     String method = "google.cloud.language.v2.LanguageService.AnalyzeSentiment";
     Key apiKey = null;

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -70,6 +70,7 @@ public class AuthExampleIT {
     Key apiKey = null;
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
+      assertNotNull(apiKey)
       System.out.println("created key " + apiKey.getKeyString());
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
 

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -72,12 +72,12 @@ public class AuthExampleIT {
     try {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
       assertNotNull(apiKey);
-      assert
+      System.out.println("key string " + apiKey.getKeyString());
       String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
-
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
+        System.out.println("trying to delete " + apiKey.getKeyString());
        // AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -21,6 +21,7 @@ import static org.junit.Assert.assertTrue;
 
 import com.google.api.apikeys.v2.Key;
 import com.google.cloud.ServiceOptions;
+import io.grpc.StatusRuntimeException;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.PrintStream;
@@ -90,7 +91,7 @@ public class AuthExampleIT {
     for (int i = 0; i < retries; i++) {
       try {
         return ApiKeyAuthExample.authenticateUsingApiKey(apiKey);
-      } catch (IOException e) {
+      } catch (StatusRuntimeException e) {
         if (e.getMessage().contains("API key expired")) {
           System.out.println("API key not yet active, retrying...");
           try {

--- a/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
+++ b/auth/src/test/java/com/google/cloud/auth/samples/AuthExampleIT.java
@@ -73,13 +73,35 @@ public class AuthExampleIT {
       apiKey = AuthTestUtils.createTestApiKey(projectId, keyDisplayName, service, method);
       assertNotNull(apiKey);
       System.out.println("key string " + apiKey.getKeyString());
-      String output = ApiKeyAuthExample.authenticateUsingApiKey(apiKey.getKeyString());
+      String output = authenticateUsingApiKeyWithRetry(apiKey.getKeyString());
       assertTrue(output.contains("magnitude:"));
     } finally {
       if (apiKey != null) {
         System.out.println("trying to delete " + apiKey.getKeyString());
-       // AuthTestUtils.deleteTestApiKey(apiKey.getName());
+        AuthTestUtils.deleteTestApiKey(apiKey.getName());
       }
     }
+  }
+
+  static String authenticateUsingApiKeyWithRetry(String apiKey) throws IOException {
+    int retries = 5;
+    int delay = 2000; // 2 seconds
+
+    for (int i = 0; i < retries; i++) {
+      try {
+        return ApiKeyAuthExample.authenticateUsingApiKey(apiKey);
+      } catch (IOException e) {
+        if (e.getMessage().contains("API key expired")) {
+          System.out.println("API key not yet active, retrying...");
+          try {
+            Thread.sleep(delay);
+          } catch (InterruptedException ignored) {}
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    throw new IOException("API key never became active after retries.");
   }
 }

